webnn/resources/test_data/hard_swish.json

```diff
@@ -0,0 +1,334 @@
+{
+  "tests": [
+    {
+      "name": "hardSwish float32 1D tensor",
+      "inputs": {
+        "x": {
+          "shape": [24],
+          "data": [
+            0.7341583533045579,
+            9.118859151005996,
+            3.545238531520827,
+            2.621943879280181,
+            -6.445507690595167,
+            -1.6835596550754381,
+            5.523179785756591,
+            -5.958856051028132,
+            -9.169189933081544,
+            6.420943542920213,
+            -3.293031330275471,
+            1.0410166785810624,
+            -7.246322671816956,
+            -0.9472730969847909,
+            -5.778352255817807,
+            3.185229125228698,
+            -7.261818072290236,
+            4.174602615173795,
+            3.7802628241590686,
+            -6.07124038718255,
+            -9.909919471919547,
+            -7.744259390113584,
+            -8.286120816748381,
+            8.083491160956697
+          ],
+          "type": "float32"
+        }
+      },
+      "expected": {
+        "name": "output",
+        "shape": [24],
+        "data": [
+          0.4569105803966522,
+          9.11885929107666,
+          3.545238494873047,
+          2.4567370414733887,
+          0,
```

Comment:
Fail to run these **hardSwish** tests by Chromium integrated WebNN API with XNNPack API back,  since XNNPack API can distinguish **0.0** and **-0.0**.
@fdwr @huningxin Any suggestion? Thanks.

Reply:
Signed and unsigned zero can be treated as equivalent (IEEE does https://en.wikipedia.org/wiki/Signed_zero#Comparisons), and I'm pretty sure that we do (will verify Monday). I recommend that before checking for ULP differences in `assert_array_approx_equals_ulp`, first check if `expected == actual`, and only if they fail, would you need to check ULP too.

Reply:
@fdwr Thanks for your suggestion! Yes, if actual was  as same as expect one, no need to measure the ULP distance which would save testing time. Please take another look, thanks.